src/lib/cli.ts

```diff
@@ -0,0 +1,103 @@
+import { app } from 'electron';
+import { executeWPCli } from '../../vendor/wp-now/src/execute-wp-cli';
+
+type CommandAction = 'wp';
+interface Command {
+	action: CommandAction;
+	args: string[];
+	executed: boolean;
+}
+
+const STANDALONE_COMMANDS: [ CommandAction ] = [ 'wp' ];
+const systemLog = console.log;
+let cliCommand: Command;
+
+export const isCLI = () => app.commandLine.hasSwitch( 'cli' );
+
+export const getCLIDataForMainInstance = () => {
+	if ( ! isCLI() ) {
+		return {};
+	}
+
+	return { cliCommand: app.commandLine.getSwitchValue( 'cli' ) };
+};
+
+export const listenCLICommands = () => {
+	app.on( 'second-instance', async ( event, commandLine, workingDirectory, additionalData ) => {
+		const data = additionalData as { cliCommand: string };
+		setCommand( data.cliCommand );
+		await executeCLICommand();
+	} );
+};
+
+export const processCLICommand = async ( {
+	mainInstance,
+	appBoot,
+}: {
+	mainInstance: boolean;
+	appBoot?: () => Promise< void >;
+} ) => {
+	setCommand( app.commandLine.getSwitchValue( 'cli' ) );
+
+	if ( STANDALONE_COMMANDS.includes( cliCommand.action ) ) {
+		await executeCLICommand();
+		app.quit();
+	} else if ( mainInstance && appBoot ) {
+		await appBoot();
+	} else {
+		app.quit();
+	}
+};
+
+export const executeCLICommand = async () => {
+	if ( ! cliCommand || cliCommand.executed ) {
+		return false;
+	}
+	cliCommand.executed = true;
+
+	await commands[ cliCommand.action ]?.();
+	return true;
+};
+
+const commands = {
+	wp: async () => {
+		disableConsole();
```

Comment:
It might be interesting to rewire console logs to the [logging system](https://github.com/Automattic/studio/blob/4320082e814b586a8b57bfde930e706ae79f3d97/src/logging.ts) instead of disabling it. This will facilitate debugging potential issues on CLI commands.

Reply:
Let's explore this option in a separate PR.